src/Reflection/ReflectionEnumCase.php

```diff
@@ -22,13 +23,52 @@
 
 class ReflectionEnumCase
 {
+    /** @var non-empty-string */
+    private string $name;
+
+    private Node\Expr|null $value;
+
+    /** @var list<ReflectionAttribute> */
+    private array $attributes;
+
+    private string $docComment;
+
+    /** @var positive-int */
+    private int $startLine;
+
+    /** @var positive-int */
+    private int $endLine;
+
+    /** @var positive-int */
+    private int $startColumn;
+
+    /** @var positive-int */
+    private int $endColumn;
+
     private CompiledValue|null $compiledValue = null;
 
     private function __construct(
         private Reflector $reflector,
-        private EnumCase $node,
+        EnumCase $node,
         private ReflectionEnum $enum,
     ) {
+        /** @var non-empty-string $name */
```

Comment:
Hmm, can our CS tooling replace some of these with assertions?

Good enough for now, just don't like having a leap of faith

Reply:
Probably in the next release, I know the author 😄